privacy_evaluator/attacks/membership_inference/membership_inference.py

```diff
@@ -0,0 +1,111 @@
+from typing import Tuple, Dict
+import numpy as np
+
+from privacy_evaluator.metrics.basics import accuracy
+from privacy_evaluator.attacks.attack import Attack
+from privacy_evaluator.classifiers.classifier import Classifier
+
+
+class MembershipInferenceAttack(Attack):
+    """MembershipInferenceAttack base class."""
+
+    def __init__(
+        self,
+        target_model: Classifier,
+        x_train: np.ndarray,
+        y_train: np.ndarray,
+        x_test: np.ndarray,
+        y_test: np.ndarray,
+    ):
+        """Initializes a MembershipInferenceAttack class.
+
+        :param target_model: The target model to be attacked.
+        :param x_train: Data that was used to train the target model.
+        :param y_train: Labels for the data that was used to train the target model.
+        :param x_test: Data that was not used to train the target model.
+        :param y_test: Labels for the data that was not used to train the target model.
+        """
+        super().__init__(target_model, x_train, y_train, x_test, y_test)
+
+    def attack(self, *args, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """Performs the attack on the target model.
+
+        :param args: The arguments of the attack.
+        :param kwargs: The keyword arguments of the attack.
+        :return: Result of the attack.
+        """
+        return self.infer(*args, **kwargs)
+
+    def infer(self, *args, **kwargs) -> Tuple[np.ndarray, np.ndarray]:
+        """Alias method for attack().
+
+        :param args: The arguments of the attack.
+        :param kwargs: The keyword arguments of the attack.
+        :return: Result of the attack.
+        """
+        raise NotImplementedError(
+            "Method 'infer()' needs to be implemented in subclass"
+        )
+
+    def target_model_train_accuracy(self) -> np.float32:
+        """Calculates the train accuracy of the target model.
+
+        :return: Train accuracy.
+        """
+        return accuracy(self.y_train, self.target_model.predict(self.x_train))
+
+    def target_model_test_accuracy(self) -> np.float32:
```

Comment:
are those methods that need to be used from outside? If not, use an underscore before the function name (for internal function)

--> or if they are not directly needed for the attack, think about moving them to the metric class